File: ValidParantheses.java
class ValidParantheses {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(char c : s.toCharArray()){
            if(c == '{' || c == '[' || c == '('){
                stack.push(c);
            }else{
                if(stack.isEmpty()){
                    return false;
                }else{
                    char top = stack.peek();
                    if(c == '}' && top == '{' ||
                      c == ')' && top =='(' ||
                      c == ']' && top == '['){
                        stack.pop();
                    }else{
                        return false;
                    }
                }
            }
        }
        return stack.isEmpty();
    }
    
    public static void main(String[] args) {
        System.out.println(isValid("()[]{}"));
    }
}

File: stack/valid_parantheses.cpp
#include<bits/stdc++.h>
using namespace std;

class Parantheses {
public:
    bool isValid(string s) {
        stack<char> mystack;
        for(int i=0;i<s.length();i++){
            if(mystack.empty()){
                mystack.push(s[i]);
                
            }
            else if ((mystack.top() == '(' && s[i] == ')')
                 || (mystack.top() == '{' && s[i] == '}')
                 || (mystack.top() == '[' && s[i] == ']')){
                mystack.pop();
            }
            else{
                mystack.push(s[i]);
            }
            
        }
        if(mystack.empty()){
            return true;
            
        }
        else{
            return false;
        }
    }
};
int main(){
  Parantheses p;
  string s="()[]{}";
  
  cout<<p.isValid(s);
  
  return 0;
}
